docs/bugpattern/LockOnNonEnclosingClassLiteral.md

```diff
@@ -1,18 +1,35 @@
-Lock on the class other than the enclosing class of the code block can
-unintentionally prevent the locked class or other classing using the same lock
-being used properly. A issue can be caused when the locked class being edited or
-deleted. And it can also be exhausting and time consuming for the others who is
-using the same class literal as lock to make sure the synchronized blocks can
-work properly as expected. Hence locking on the class other than the enclosing
-class of the synchronized code block is disencouraged by the error prone.
+Having a lock on a class, other than the enclosing class of the code block, can
+unintentionally prevent the locked class from being used properly. This also
+holds for other classes that are using the same lock.
+An issue can arise when the locked class is being edited or deleted.
```

Comment:
Not sure if this sentence holds any value compared to the sentences above TBH.

Reply:
Yeah makes sense, I think I tried to stay to close to the original content.